### ISSUE TITLE
[BUG] Invalid IP for foo.apps-crc.testing on crc start w/1.11.0 on Windows

### ISSUE THREAD

**Issue by ron1** [code-ready/crc#1300]
### General information

  * OS: Windows
  * Hypervisor: Hyper-V
  * Did you run `crc setup` before starting it (Yes/No)? Yes
  * Running CRC on: Laptop

## CRC version
```bash
crc version: 1.11.0+883ca49
OpenShift version: 4.4.5 (embedded in binary)  
```
  
## CRC status
```bash
DEBU Running 'C:\Users\ron1\.crc\bin\oc\oc.exe get co -ojson --kubeconfig C:\Users\ron1\.crc\machines\crc\kubeconfig --context admin --cluster crc'
DEBU Command failed: exit status 1
DEBU stdout:
DEBU stderr: error: CreateFile C:\Users\ron1\.crc\machines\crc\kubeconfig: The system cannot find the file specified.
DEBU error: CreateFile C:\Users\ron1\.crc\machines\crc\kubeconfig: The system cannot find the file specified.
ERRO Error getting config for native Go SSH: open C:\Users\ron1\.crc\machines\crc\id_rsa: The system cannot find the file specified.
```

## CRC config
```bash
# Put `crc config view` output here
```

## Host Operating System
```bash
Host Name:                 XX
OS Name:                   Microsoft Windows 10 Pro
OS Version:                10.0.19041 N/A Build 19041
OS Manufacturer:           Microsoft Corporation
OS Configuration:          Standalone Workstation
OS Build Type:             Multiprocessor Free
Registered Owner:          HP
Registered Organization:   HP
Product ID:                XX
Original Install Date:     6/3/2020, 10:56:06 PM
System Boot Time:          6/8/2020, 1:02:59 PM
System Manufacturer:       HP
System Model:              HP ENVY Laptop 17-ae1xx
System Type:               x64-based PC
Processor(s):              1 Processor(s) Installed.
                           [01]: Intel64 Family 6 Model 142 Stepping 10 GenuineIntel ~1494 Mhz
BIOS Version:              Insyde F.28, 4/29/2019
Windows Directory:         C:\WINDOWS
System Directory:          C:\WINDOWS\system32
Boot Device:               \Device\HarddiskVolume3
System Locale:             en-us;English (United States)
Input Locale:              en-us;English (United States)
Time Zone:                 (UTC-05:00) Eastern Time (US & Canada)
Total Physical Memory:     16,231 MB
Available Physical Memory: 2,528 MB
Virtual Memory: Max Size:  18,663 MB
Virtual Memory: Available: 2,442 MB
Virtual Memory: In Use:    16,221 MB
Page File Location(s):     C:\pagefile.sys
Domain:                    WORKGROUP
Logon Server:              \\XX
Hotfix(s):                 4 Hotfix(s) Installed.
                           [01]: KB4552925
                           [02]: KB4537759
                           [03]: KB4557968
                           [04]: KB4556803
Network Card(s):           5 NIC(s) Installed.
                           [01]: Cisco AnyConnect Secure Mobility Client Virtual Miniport Adapter for Windows x64
                                 Connection Name: Ethernet 2
                                 Status:          Hardware not present
                           [02]: Intel(R) Dual Band Wireless-AC 7265
                                 Connection Name: Wi-Fi
                                 DHCP Enabled:    Yes
                                 DHCP Server:     192.168.1.1
                                 IP address(es)
                                 [01]: 192.168.1.240
                           [03]: Hyper-V Virtual Ethernet Adapter
                                 Connection Name: vEthernet (Default Switch)
                                 DHCP Enabled:    No
                                 IP address(es)
                                 [01]: 172.31.48.1
                                 [02]: fe80::6d45:4802:9db3:74d6
                           [04]: Realtek PCIe GbE Family Controller
                                 Connection Name: Ethernet
                                 Status:          Media disconnected
                           [05]: Bluetooth Device (Personal Area Network)
                                 Connection Name: Bluetooth Network Connection
                                 Status:          Media disconnected
Hyper-V Requirements:      A hypervisor has been detected. Features required for Hyper-V will not be displayed.
```

### Steps to reproduce

  1. crc setup
  2. crc start

### Expected
crc start completed successfully

### Actual
crc start generated the following error
```
WARN foo.apps-crc.testing resolved to [92.242.140.21] but 172.31.56.119 was expected
ERRO Failed to query DNS from host: Invalid IP for foo.apps-crc.testing
```

### Logs
https://gist.github.com/ron1/31a0daf73a55e80bb6987174d9508e9c

**Comment by gbraad** [comment]
Was this while connected to the VPN? If so, please try this while disconnected ...

**Comment by ron1** [comment]
@gbraad This was while disconnected from the VPN.

**Comment by cfergeau** [comment]
92.242.140.21 is basically the DNS server saying it could not find `foo.apps-crc.testing` https://askubuntu.com/questions/587895/why-is-the-ip-92-242-140-21-connecting-to-one-of-my-ports-is-it-malware
This is probably an unexpected consequence of the `goodhosts` change.

**Comment by gbraad** [comment]
That is why it is puzzling, as it seems `crc.testing` worked, but the `apps-crc.testing` domain didn't?

Is this repeatable? please perform a `crc cleanup` and `crc start` again

**Comment by ron1** [comment]
@cfergeau What version of crc do you recommend I try? I do have a goodhosts binary in the bin directory. Also note that the May 2020 Windows update was recently applied to my machine.

@gbraad I have followed the cleanup and restart procedures numerous times deleting the .crc directory contents each time just to be sure.

**Comment by ron1** [comment]
BTW, I also disabled avast virus scanner and defender firewall to be sure the environment was as clean as possible.

**Comment by gbraad** [comment]
> disabled avast virus scanner

check

**Comment by gbraad** [comment]
either 0.9.0 or 0.11.0 (current latest) due to the certificates issue

**Comment by ron1** [comment]
I will try 0.9.0. Do you know if crc has been tested with latest Windows version?

**Comment by ron1** [comment]
BTW, I also have WSL2 installed but am running crc from a traditional command prompt.

**Comment by ron1** [comment]
@gbraad @cfergeau Unfortunately, crc 0.9.0 exhibits the same behavior.

Do you have any other troubleshooting suggestions?

**Comment by gbraad** [comment]
> BTW, I also have WSL2 installed but am running crc from a traditional command prompt.

both shouldn't matter.

**Comment by gbraad** [comment]
Not sure what Verizon DNS Assistance is, but if this is a ctach all that might refer you to an actual correct hostname in the case of a typo, etc I would say that this kind of catch-all edges on DNS snooping and sounds very unwanted. It is an infrastructural issue that CRC could not work around as it is a network configuration that might overrule local settings

**Comment by ron1** [comment]
@gbraad I suspect many crc users telework and rely on ISPs like Verizon for internet connectivity. As described here https://askubuntu.com/questions/587895/why-is-the-ip-92-242-140-21-connecting-to-one-of-my-ports-is-it-malware, it appears ISPs like Verizon and others provide this "DNS Assistance" as a "service". If I was able to configure crc to workaround the problem by identifying a specific nameserver such as 1.1.1.1 or 8.8.8.8 with the -n parameter, I agree a work-around would not be necessary. But identifying a specific nameserver with the -n parameter did not fix the problem in this case.

Is it possible for crc to distrust non-authoritative DNS entries, at least for the '.testing' domain?

**Comment by gbraad** [comment]
DNS servers are often geographically assigned. It would be onnodige for us to maintain a list for such services.

Providing a DNS on start can work around the ordering, but will not be default. This is an issue with Verizon as they do not honor the NXDOMAIN status to do proper cascading (they short circuit requests and disable the use of a 2nd or more nameservers)

**Comment by gbraad** [comment]
Startup check depends on the return of NXDOMAIN to determine if settings are incorrect. We might bd able to enhance this to include an 'equals to CRC ip', ... @code-ready/crc-devel WDYT?

**Comment by gbraad** [comment]
Th

**Comment by cfergeau** [comment]
> Startup check depends on the return of NXDOMAIN to determine if settings are incorrect. ~We might bd able to enhance this to include an 'equals to CRC ip', ... @code-ready/crc-devel WDYT?~ This already works this way ...

We may be able to detect DNS hijacking by trying to resolve `crc.invalid`, which is guaranteed to return `NXDOMAIN` according to https://tools.ietf.org/html/rfc2606#page-2 
Hopefully verizon did not special case `.invalid` and redirects to some webpage of their own in this case as well.

**Comment by NickMRamirez** [comment]
I was getting this same problem on Linux. The problem is that OpenShift Local adds an entry to `/etc/hosts`. (On Windows, it would be `C:\Windows\System32\drivers\etc\hosts`) For example:

```
192.168.130.11 api.crc.testing canary-openshift-ingress-canary.apps-crc.testing console-openshift-console.apps-crc.testing default-route-openshift-image-registry.apps-crc.testing downloads-openshift-console.apps-crc.testing oauth-openshift.apps-crc.testing
```

But this entry is missing `foo.apps-crc.testing`.

To resolve this, add this domain to the entry in `/etc/hosts`. Stop and start OpenShift Local:

```
$ crc stop
$ crc start
```

Is it possible to re-open this bug so that the hosts entry can be fixed?

**Comment by praveenkumar** [comment]
It was warning and fixed sometime back, make sure you use the latest version of CRC in windows.

**Comment by NickMRamirez** [comment]
@praveenkumar Should I open a new issue? This is on Linux using the following version: 

```
$ crc version
CRC version: 2.20.0+f3a947
OpenShift version: 4.13.0
Podman version: 4.4.4

```

This causes the OpenShift Local installation to fail.

**Comment by cfergeau** [comment]
Yes, better to open a new issue with all the needed details.
